src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/MetricsCollectorSpark.java

```diff
@@ -0,0 +1,116 @@
+package org.broadinstitute.hellbender.tools.spark.pipelines.metrics;
+
+import htsjdk.samtools.SAMFileHeader;
+
+import htsjdk.samtools.metrics.Header;
+import org.apache.spark.api.java.JavaRDD;
+import org.broadinstitute.hellbender.engine.AuthHolder;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.metrics.MetricsArgs;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Interface implemented by Spark metrics collectors to allow them
+ * to run as either a standalone Spark tool, or under the control of
+ * CollectMultipleMetricsSpark, which reuses the same input RDD to run
+ * multiple collectors.
+ *
+ * Spark collectors should be factored into the following implementation
+ * classes:
+ *
+ * <li>
+ * A class derived from MetricsArgs that defines the input arguments for
+ * the collector in a form that is suitable as a command line argument
+ * collection.
+ * A class that implements this interface, parameterized with the
+ * MetricsArgs-derived class.
+ * </li>
+ *
+ * These two classes can then be used either in the standalone collector tool (see
+ * CollectInsertSizeMetricsSpark as an example), or from CollectMultipleMetricsSpark.
+ *
+ * The general lifecyle of a Spark collector looks like this:
+ *
+ *     CollectorType collector = new CollectorType<CollectorArgType>()
+ *     CollectorArgType args = // may call collector.gatherInputArguments();
+ *
+ *     // pass the input arguments back to the collector for initialization
+ *     collector.initializeCollector(args);
+ *
+ *     ReadFilter filter == collector.getReadFilter();
+ *     collector.collectMetrics(
+ *         getReads().filter(filter),
+ *         getHeaderForReads(),
+ *         getReadSourceName(),
+ *         getAuthHolder()
+ *     );
+ *     collector.finishCollection();
```

Comment:
The result of the metrics collection is an RDD that will usually be reduced to a single object whos' type is specific to the collector (i.e. a QualityYieldMetrics object, or a MultiLevelCollector/derivative) that is serialized out to a file by the collector when the finish method is called. (I think the two implementations in this PR currently do that as part of collectMetrics). We could change this to have the result returned to the caller/driver such as CollectMultipleMetricsSpark, but without any common type ancestor I'm not sure what it could do with it other than pass it back to finish for export. Maybe thats better just because its more explicit. I'm open to suggestions.